tests/python_package_test/test_dask.py

```diff
@@ -1659,6 +1659,19 @@ def test_dask_classes_and_sklearn_equivalents_have_identical_constructors_except
     assert dask_spec.defaults[-1] is None
 
 
+@pytest.mark.parametrize(
+    "dask_class",
+    [
+        lgb.DaskLGBMClassifier,
+        lgb.DaskLGBMRegressor,
+        lgb.DaskLGBMRanker
+    ]
+)
+def test_dask_early_stopping_raises_not_implemented_error(dask_class):
+    with pytest.raises(NotImplementedError):
```

Comment:
Wherever you use `pytest.raises()` in this project, please use `match` to ensure the code is raising **exactly** the error message you're expecting. That improves the chances that other unrelated errors could be caught by this test.

In case you're unfamiliar with `pytest.raises()`, here are the docs:  https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-raises

And here's an example of the pattern I'm requesting:

https://github.com/microsoft/LightGBM/blob/7a801f7b7b1ed8faf4cd1dbc09086c269fc51e1a/tests/python_package_test/test_basic.py#L781-L788

Reply:
This test has been removed. In future usages of `pytest.raises` I will also set `match`.

Reply:
thank you